solr/core/src/java/org/apache/solr/cli/CreateTool.java

```diff
@@ -167,10 +166,9 @@ protected void createCore(CommandLine cli, SolrClient solrClient) throws Excepti
 
     String coreRootDirectory; // usually same as solr home, but not always
 
-    Map<String, Object> systemInfo =
-        solrClient
-            .request(new GenericSolrRequest(SolrRequest.METHOD.GET, CommonParams.SYSTEM_INFO_PATH))
-            .asMap();
```

Comment:
You are right that I don't like NamedList, it's always been a weird Solr specific data structure.  Why did we feel the need to come up with a unique data structure that isn't in the JDK?    Having said that, what you are proposing is clearly an improvement!

Reply:
I also wish we just had a cleaner data structure for these GSR that return JSON.  I love that in our tests we have `assertJQ` and `JQ` methods that lets us grab some json.   In a perfect world, this call would have returned a JSON datastructure, and I would use a Json query line to pluck out the `core_root`.   This would be more compelling if we had  a much more nested Map data structure to go through!

Reply:
I think @gerlowskija would argue that the "perfect world" (we're slowly iterating towards) would not be JSON at this level of abstraction, it'd be a a machine-generated POJO generated from our OpenAPI that you call normal typed methods on.  Correct me if I'm wrong Jason; I agree with that vision too.

But in the meantime Eric, get to know NamedList.  It's not going anywhere anytime soon, and it's also not hard to use IMO.  I wish it had the convenience methods that SolrParams has but not a big deal.

Reply:
+1 to the POJO!   And yep.